src/librustc_passes/dead.rs

```diff
@@ -569,16 +569,26 @@ impl Visitor<'tcx> for DeadVisitor<'tcx> {
 
     fn visit_item(&mut self, item: &'tcx hir::Item) {
         if self.should_warn_about_item(item) {
-            // For items that have a definition with a signature followed by a
-            // block, point only at the signature.
+            // For most items, we want to highlight its identifier
             let span = match item.kind {
                 hir::ItemKind::Fn(..) |
                 hir::ItemKind::Mod(..) |
                 hir::ItemKind::Enum(..) |
                 hir::ItemKind::Struct(..) |
                 hir::ItemKind::Union(..) |
                 hir::ItemKind::Trait(..) |
-                hir::ItemKind::Impl(..) => self.tcx.sess.source_map().def_span(item.span),
+                hir::ItemKind::Impl(..) => {
+                    // FIXME(66095): Because item.span is annotated with things
+                    // like a macro_backtrace, and ident.span isn't, we use the
+                    // def_span method if it's part of a macro invocation
+                    // We should probably annotate ident.span with the macro
+                    // context, but that's a larger change.
+                    if item.span.macro_backtrace().len() == 0 {
```

Comment:
would `item.span.source_callee().is_none()` work here instead of building the macro backtrace?

Reply:
I'll give it a try.  Can you explain what source_callee is doing for my own education?

Reply:
If a span is inside a macro, they have expansion data, including among other things where the span for the macro call is coming from, which `source_callee` returns. If a span has no expansion data, it means that it isn't coming from a macro context.